### PR TITLE
Update symbol

### PR DESCRIPTION
**Pull request by jaortiz**

### CODE REVIEW

**Comment by jaortiz** [comment]
I made some changes to buildExports.
- Made more generic so it doesn't just use icon naming
- Added a check to see if icon/symbol folder exists and create folder + package.json if need be

File: components/symbol/examples/40-local-tokens.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import {
+	MastercardAcceptedSymbol,
+	BPayLandSymbol,
+	AppleStoreSymbol,
+	WBCLogo,
+	STGLogo,
+} from '@westpac/symbol';
+import { Cell, Grid, Name } from './_utils';
+
+const Wrapper = ({ children, ...props }) => (
```
Comment (dominikwilkowski) [inline components/symbol/examples/40-local-tokens.js]:
```suggestion
const Wrapper = ({ children, symbol, ...props }) => (
```

File: components/symbol/examples/40-local-tokens.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	<div
+		css={{
+			border: '2px solid red',
+			padding: '24px',
+			height: '100%',
+		}}
+	>
+		{children}
```
Comment (dominikwilkowski) [inline components/symbol/examples/40-local-tokens.js]:
```suggestion
		{children} - {symbol}
```

Comment (dominikwilkowski) [reply components/symbol/examples/40-local-tokens.js]: Let's show in the examples that you have access to this `symbol` prop now in the `Wrapper`.

File: components/symbol/src/Symbol.js [hunk 3]
```diff
@@ -49,8 +50,16 @@ export const Symbol = ({
 	children,
 	...props
 }) => {
+	const { [pkg.name]: overridesWithTokens } = useBrand();
+
+	const overrides = {
+		Wrapper: SymbolWrapper,
+		svgAttributes: {},
```
Comment (dominikwilkowski) [inline components/symbol/src/Symbol.js]: We also need to spread css for each component. I forgot that in `Icon` as well.

```suggestion
		svgAttributes: {},
		css {},
```

File: components/symbol/src/Symbol.js [hunk 4]
```diff
@@ -64,10 +73,11 @@ export const Symbol = ({
 				role="img"
 				focusable="false"
 				style={{ width: '100%', height: '100%' }}
+				{...overrides.svgAttributes}
 			>
```
Comment (dominikwilkowski) [inline components/symbol/src/Symbol.js]:
```suggestion
			css={overrides.css}
			>
```

File: helpers/buildExports.js [hunk 5]
```diff
@@ -6,28 +6,31 @@ const fs = require('fs');
 /**
  * Getting all files from a directory
  *
- * @param  {string} iconPath - The path to the folder
+ * @param  {string} svgPath - The path to the folder containing all component svgs
+ *
```
Comment (dominikwilkowski) [inline helpers/buildExports.js]:
```suggestion
```

File: helpers/buildExports.js [hunk 6]
```diff
@@ -6,28 +6,31 @@ const fs = require('fs');
...
+ * @param	{string} component - The component name i.e. icons, symbols
```
Comment (dominikwilkowski) [inline helpers/buildExports.js]:
```suggestion
 * @param  {string} component - The component name i.e. icons, symbols
```

File: helpers/buildExports.js [hunk 7]
```diff
@@ -6,28 +6,31 @@ const fs = require('fs');
...
  *
  * @return {array}           - an array of all files inside a folder with the ".js" extension stripped
```
Comment (dominikwilkowski) [inline helpers/buildExports.js]:
```suggestion
 * @return {array}                       - an array of all files inside a folder with the ".js" extension stripped
```

File: helpers/buildExports.js [hunk 8]
```diff
@@ -6,28 +6,31 @@ const fs = require('fs');
 /**
  * Getting all files from a directory
  *
- * @param  {string} iconPath - The path to the folder
+ * @param  {string} svgPath - The path to the folder containing all component svgs
```
Comment (dominikwilkowski) [inline helpers/buildExports.js]:
```suggestion
 * @param  {string} svgPath      - The path to the folder containing all component svgs
```

File: helpers/buildExports.js [hunk 9]
```diff
@@ -6,28 +6,31 @@ const fs = require('fs');
...
+	console.info(chalk.green(`✅ Got all ${component} successfully`));
+	return svgs;
 }
 
 /**
- * Insert an array of icons into a js export file
+ * Insert an array of component svgs into a js export file
  *
- * @param  {array}  icons     - An array of icons
+ * @param  {array}  svgs     - An array of component svgs
```
Comment (dominikwilkowski) [inline helpers/buildExports.js]:
```suggestion
 * @param  {array}  svgs      - An array of component svgs
```

File: helpers/buildExports.js [hunk 10]
```diff
@@ -59,38 +62,72 @@ function writePkg(pkgPath, content) {
 }
 
 /**
- * Insert an array of icons into the preconstruct entrypoint array of a package.json
+ * Insert an array of component svgs into the preconstruct entrypoint array of a package.json
  *
- * @param  {array}  icons   - An array of icons
+ * @param  {array}  svgs   - An array of component svgs
```
Comment (dominikwilkowski) [inline helpers/buildExports.js]:
```suggestion
 * @param  {array}  svgs    - An array of component svgs
```

File: helpers/buildExports.js [hunk 11]
```diff
@@ -59,38 +62,72 @@ function writePkg(pkgPath, content) {
...
 
 	console.info(chalk.green('✅ package.json file written successfully'));
 }
 
 /**
- * Fix the source key inside each icons package.json file
+ * Fix the source key inside each svgs package.json file
  *
- * @param  {array} icons - An array of all icons
+ * @param  {array} svgs - An array of all component svgs
```
Comment (dominikwilkowski) [inline helpers/buildExports.js]:
```suggestion
 * @param  {array} svgs  - An array of all component svgs
```

File: helpers/buildExports.js [hunk 12]
```diff
@@ -59,38 +62,72 @@ function writePkg(pkgPath, content) {
...
-			pkg.preconstruct = {};
+function fixSource(svgs, component) {
+	svgs.map(svg => {
+		// check if package exists
+		const folderPath = path.normalize(`${process.cwd()}/${svg}`);
+		if (!fs.existsSync(folderPath)) {
+			// have a log about creating folder here
+			// need to have error handlers here
+			try {
+				fs.mkdirSync(folderPath);
```
Comment (dominikwilkowski) [inline helpers/buildExports.js]: All of this is done inside `preconstruct` and wasn't needed here :)

File: helpers/buildExports.js [hunk 13]
```diff
@@ -59,38 +62,72 @@ function writePkg(pkgPath, content) {
...
+			} catch (error) {
+				console.error(chalk.red(`❌ ${error}`));
+				process.exit(1);
+			}
+
+			const package = {
+				main: `dist/${component.slice(0, -1)}.cjs.js`,
```
Comment (dominikwilkowski) [inline helpers/buildExports.js]: I think we need to be careful with duplicating this in `preconstruct` and here... `preconstruct` might change this in the future.

File: helpers/buildExports.js (hunk 12 above)
Comment (jaortiz) [reply helpers/buildExports.js]: It wasn't creating folders for me before thats why I added it in. Ill check with new version

Comment (dominikwilkowski) [reply helpers/buildExports.js]: Yeah that's why I said we need to run it twice. It does create it even in the old version but you need to run it first.

Comment (jaortiz) [reply helpers/buildExports.js]: I removed this part, removed one of the symbol folders and ran the build script and it errors out like before
```
Error: Cannot find module '/Users/Jeremy/Dev/gel-symbol/components/symbol/AmericanExpressSymbol/package.json'
```

Comment (dominikwilkowski) [reply helpers/buildExports.js]: Did yoiu run `preconstruct`?

Comment (jaortiz) [reply helpers/buildExports.js]: ahhh I was always running preconstruct dev not build. All good I'll remove it.

Comment (dominikwilkowski) [reply helpers/buildExports.js]: sorry that's my fault for not communicating properly... 😁

**Pull request merged**